commit 2c14af93e1902751db5c7ca17182e0288e882b10
Author: MH <80083951+zanac1986@users.noreply.github.com>
Date:   Mon Dec 5 23:42:48 2022 +0800

    [alertmanager] Refactor (#2781)
    
    * [alertmanager] Refactor
    
    [alertmanager] Refactor
    
    Signed-off-by: zanac1986 <zanhsieh@protonmail.com>
    
    * Update charts/alertmanager/README.md
    
    Signed-off-by: MH <zanhsieh@gmail.com>
    
    Signed-off-by: zanac1986 <zanhsieh@protonmail.com>
    Signed-off-by: MH <zanhsieh@gmail.com>
    Co-authored-by: MH <zanhsieh@gmail.com>

diff --git a/charts/alertmanager/Chart.yaml b/charts/alertmanager/Chart.yaml
index 1468ce5e..628e9b38 100644
--- a/charts/alertmanager/Chart.yaml
+++ b/charts/alertmanager/Chart.yaml
@@ -6,7 +6,7 @@ icon: https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/a
 sources:
   - https://github.com/prometheus/alertmanager
 type: application
-version: 0.22.0
+version: 0.22.1
 appVersion: v0.24.0
 maintainers:
   - name: monotek
diff --git a/charts/alertmanager/README.md b/charts/alertmanager/README.md
index 7daae41d..ef0e1db3 100644
--- a/charts/alertmanager/README.md
+++ b/charts/alertmanager/README.md
@@ -10,14 +10,14 @@ As per [prometheus.io documentation](https://prometheus.io/docs/alerting/latest/
 
 Kubernetes 1.14+
 
-## Get Repo Info
+## Get Repository Info
 
 ```console
 helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
 helm repo update
 ```
 
-_See [helm repo](https://helm.sh/docs/helm/helm_repo/) for command documentation._
+_See [`helm repo`](https://helm.sh/docs/helm/helm_repo/) for command documentation._
 
 ## Install Chart
 
diff --git a/charts/alertmanager/templates/_helpers.tpl b/charts/alertmanager/templates/_helpers.tpl
index 47d01ca1..4d73e01e 100644
--- a/charts/alertmanager/templates/_helpers.tpl
+++ b/charts/alertmanager/templates/_helpers.tpl
@@ -3,8 +3,8 @@
 Expand the name of the chart.
 */}}
 {{- define "alertmanager.name" -}}
-{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
-{{- end -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- end }}
 
 {{/*
 Create a default fully qualified app name.
@@ -12,24 +12,24 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 If release name contains chart name it will be used as a full name.
 */}}
 {{- define "alertmanager.fullname" -}}
-{{- if .Values.fullnameOverride -}}
-{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
-{{- else -}}
-{{- $name := default .Chart.Name .Values.nameOverride -}}
-{{- if contains $name .Release.Name -}}
-{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
-{{- else -}}
-{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
-{{- end -}}
-{{- end -}}
-{{- end -}}
+{{- if .Values.fullnameOverride }}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default .Chart.Name .Values.nameOverride }}
+{{- if contains $name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
+{{- end }}
+{{- end }}
+{{- end }}
 
 {{/*
 Create chart name and version as used by the chart label.
 */}}
 {{- define "alertmanager.chart" -}}
-{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}
-{{- end -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
+{{- end }}
 
 {{/*
 Common labels
@@ -37,11 +37,11 @@ Common labels
 {{- define "alertmanager.labels" -}}
 helm.sh/chart: {{ include "alertmanager.chart" . }}
 {{ include "alertmanager.selectorLabels" . }}
-{{- if .Chart.AppVersion }}
-app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- with .Chart.AppVersion }}
+app.kubernetes.io/version: {{ . | quote }}
 {{- end }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
-{{- end -}}
+{{- end }}
 
 {{/*
 Selector labels
@@ -49,15 +49,39 @@ Selector labels
 {{- define "alertmanager.selectorLabels" -}}
 app.kubernetes.io/name: {{ include "alertmanager.name" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
-{{- end -}}
+{{- end }}
 
 {{/*
 Create the name of the service account to use
 */}}
 {{- define "alertmanager.serviceAccountName" -}}
-{{- if .Values.serviceAccount.create -}}
-    {{ default (include "alertmanager.fullname" .) .Values.serviceAccount.name }}
-{{- else -}}
-    {{ default "default" .Values.serviceAccount.name }}
-{{- end -}}
-{{- end -}}
+{{- if .Values.serviceAccount.create }}
+{{- default (include "alertmanager.fullname" .) .Values.serviceAccount.name }}
+{{- else }}
+{{- default "default" .Values.serviceAccount.name }}
+{{- end }}
+{{- end }}
+
+{{/*
+Define Ingress apiVersion
+*/}}
+{{- define "alertmanager.ingress.apiVersion" -}}
+{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion }}
+{{- printf "networking.k8s.io/v1" }}
+{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion }}
+{{- printf "networking.k8s.io/v1beta1" }}
+{{- else }}
+{{- printf "extensions/v1beta1" }}
+{{- end }}
+{{- end }}
+
+{{/*
+Define Pdb apiVersion
+*/}}
+{{- define "alertmanager.pdb.apiVersion" -}}
+{{- if $.Capabilities.APIVersions.Has "policy/v1/PodDisruptionBudget" }}
+{{- printf "policy/v1" }}
+{{- else }}
+{{- printf "policy/v1beta1" }}
+{{- end }}
+{{- end }}
\ No newline at end of file
diff --git a/charts/alertmanager/templates/ingress.yaml b/charts/alertmanager/templates/ingress.yaml
index efc9599c..fa022955 100644
--- a/charts/alertmanager/templates/ingress.yaml
+++ b/charts/alertmanager/templates/ingress.yaml
@@ -1,18 +1,12 @@
-{{- if .Values.ingress.enabled -}}
-{{- $fullName := include "alertmanager.fullname" . -}}
-{{- $svcPort := .Values.service.port -}}
+{{- if .Values.ingress.enabled }}
+{{- $fullName := include "alertmanager.fullname" . }}
+{{- $svcPort := .Values.service.port }}
 {{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
-  {{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
-  {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
-  {{- end }}
+{{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
+{{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
 {{- end }}
-{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
-apiVersion: networking.k8s.io/v1
-{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
-apiVersion: networking.k8s.io/v1beta1
-{{- else -}}
-apiVersion: extensions/v1beta1
 {{- end }}
+apiVersion: {{ include "alertmanager.ingress.apiVersion" . }}
 kind: Ingress
 metadata:
   name: {{ $fullName }}
diff --git a/charts/alertmanager/templates/pdb.yaml b/charts/alertmanager/templates/pdb.yaml
index a4654752..327a8a70 100644
--- a/charts/alertmanager/templates/pdb.yaml
+++ b/charts/alertmanager/templates/pdb.yaml
@@ -1,17 +1,13 @@
-{{- if .Values.podDisruptionBudget -}}
-{{ if $.Capabilities.APIVersions.Has "policy/v1/PodDisruptionBudget" -}}
-apiVersion: policy/v1
-{{- else -}}
-apiVersion: policy/v1beta1
-{{- end }}
+{{- if .Values.podDisruptionBudget }}
+apiVersion: {{ include "alertmanager.pdb.apiVersion" . }}
 kind: PodDisruptionBudget
 metadata:
-  name: {{ template "alertmanager.fullname" . }}
+  name: {{ include "alertmanager.fullname" . }}
   labels:
     {{- include "alertmanager.labels" . | nindent 4 }}
 spec:
   selector:
     matchLabels:
       {{- include "alertmanager.selectorLabels" . | nindent 6 }}
-{{ toYaml .Values.podDisruptionBudget | indent 2 }}
-{{- end -}}
+  {{- toYaml .Values.podDisruptionBudget | indent 2 }}
+{{- end }}
diff --git a/charts/alertmanager/templates/serviceaccount.yaml b/charts/alertmanager/templates/serviceaccount.yaml
index 9ca80f4c..1f7b8d52 100644
--- a/charts/alertmanager/templates/serviceaccount.yaml
+++ b/charts/alertmanager/templates/serviceaccount.yaml
@@ -1,4 +1,4 @@
-{{- if .Values.serviceAccount.create -}}
+{{- if .Values.serviceAccount.create }}
 apiVersion: v1
 kind: ServiceAccount
 metadata:
@@ -9,4 +9,4 @@ metadata:
   annotations:
     {{- toYaml . | nindent 4 }}
   {{- end }}
-{{- end -}}
+{{- end }}
diff --git a/charts/alertmanager/templates/services.yaml b/charts/alertmanager/templates/services.yaml
index 1b1d1dcf..8b2d2f1f 100644
--- a/charts/alertmanager/templates/services.yaml
+++ b/charts/alertmanager/templates/services.yaml
@@ -4,21 +4,21 @@ metadata:
   name: {{ include "alertmanager.fullname" . }}
   labels:
     {{- include "alertmanager.labels" . | nindent 4 }}
-{{- if .Values.service.annotations }}
+  {{- with .Values.service.annotations }}
   annotations:
-    {{- toYaml .Values.service.annotations | nindent 4 }}
-{{- end }}
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
 spec:
   type: {{ .Values.service.type }}
-{{- if .Values.service.loadBalancerIP }}
-  loadBalancerIP: {{ .Values.service.loadBalancerIP }}
-{{- end }}
-{{- if .Values.service.loadBalancerSourceRanges }}
+  {{- with .Values.service.loadBalancerIP }}
+  loadBalancerIP: {{ . }}
+  {{- end }}
+  {{- with .Values.service.loadBalancerSourceRanges }}
   loadBalancerSourceRanges:
-  {{- range $cidr := .Values.service.loadBalancerSourceRanges }}
+  {{- range $cidr := . }}
     - {{ $cidr }}
   {{- end }}
-{{- end }}
+  {{- end }}
   ports:
     - port: {{ .Values.service.port }}
       targetPort: http
diff --git a/charts/alertmanager/templates/statefulset.yaml b/charts/alertmanager/templates/statefulset.yaml
index 852a28ed..bef59a9c 100644
--- a/charts/alertmanager/templates/statefulset.yaml
+++ b/charts/alertmanager/templates/statefulset.yaml
@@ -1,14 +1,14 @@
-{{- $svcClusterPort := .Values.service.clusterPort -}}
+{{- $svcClusterPort := .Values.service.clusterPort }}
 apiVersion: apps/v1
 kind: StatefulSet
 metadata:
   name: {{ include "alertmanager.fullname" . }}
   labels:
     {{- include "alertmanager.labels" . | nindent 4 }}
-{{- if .Values.statefulSet.annotations }}
+  {{- with .Values.statefulSet.annotations }}
   annotations:
-    {{ toYaml .Values.statefulSet.annotations | nindent 4 }}
-{{- end }}
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
 spec:
   replicas: {{ .Values.replicaCount }}
   selector:
@@ -19,44 +19,44 @@ spec:
     metadata:
       labels:
         {{- include "alertmanager.selectorLabels" . | nindent 8 }}
-{{- if .Values.podLabels }}
-        {{ toYaml .Values.podLabels | nindent 8 }}
-{{- end }}
+        {{- with .Values.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
       annotations:
-      {{- if not .Values.configmapReload.enabled }}
+        {{- if not .Values.configmapReload.enabled }}
         checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
-      {{- end }}
-{{- if .Values.podAnnotations }}
-        {{- toYaml .Values.podAnnotations | nindent 8 }}
-{{- end }}
+        {{- end }}
+        {{- with .Values.podAnnotations }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
     spec:
-    {{- with .Values.imagePullSecrets }}
+      {{- with .Values.imagePullSecrets }}
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
-    {{- end }}
+      {{- end }}
       serviceAccountName: {{ include "alertmanager.serviceAccountName" . }}
-    {{- with .Values.dnsConfig }}
+      {{- with .Values.dnsConfig }}
       dnsConfig:
         {{- toYaml . | nindent 8 }}
-    {{- end }}
-    {{- with .Values.nodeSelector }}
+      {{- end }}
+      {{- with .Values.nodeSelector }}
       nodeSelector:
         {{- toYaml . | nindent 8 }}
       {{- end }}
-    {{- if or .Values.podAntiAffinity .Values.affinity }}
+      {{- if or .Values.podAntiAffinity .Values.affinity }}
       affinity:
-    {{- end }}
-    {{- if .Values.affinity }}
-        {{- toYaml .Values.affinity | nindent 8 }}
-    {{- end }}
-    {{- if eq .Values.podAntiAffinity "hard" }}
+      {{- end }}
+        {{- with .Values.affinity }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+        {{- if eq .Values.podAntiAffinity "hard" }}
         podAntiAffinity:
           requiredDuringSchedulingIgnoredDuringExecution:
             - topologyKey: {{ .Values.podAntiAffinityTopologyKey }}
               labelSelector:
                 matchExpressions:
                   - {key: app.kubernetes.io/name, operator: In, values: [{{ include "alertmanager.name" . }}]}
-    {{- else if eq .Values.podAntiAffinity "soft" }}
+        {{- else if eq .Values.podAntiAffinity "soft" }}
         podAntiAffinity:
           preferredDuringSchedulingIgnoredDuringExecution:
             - weight: 100
@@ -65,14 +65,15 @@ spec:
                 labelSelector:
                   matchExpressions:
                     - {key: app.kubernetes.io/name, operator: In, values: [{{ include "alertmanager.name" . }}]}
-    {{- end }}
-    {{- with .Values.topologySpreadConstraints }}
-      topologySpreadConstraints: {{- toYaml . | nindent 8 }}
-    {{- end }}
-    {{- with .Values.tolerations }}
+        {{- end }}
+      {{- with .Values.topologySpreadConstraints }}
+      topologySpreadConstraints:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.tolerations }}
       tolerations:
         {{- toYaml . | nindent 8 }}
-    {{- end }}
+      {{- end }}
       securityContext:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
       containers:
@@ -85,9 +86,9 @@ spec:
             - --webhook-url=http://127.0.0.1:{{ .Values.service.port }}/-/reload
           resources:
             {{- toYaml .Values.configmapReload.resources | nindent 12 }}
-          {{- if .Values.configmapReload.containerPort }}
+          {{- with .Values.configmapReload.containerPort }}
           ports:
-            - containerPort: {{ .Values.configmapReload.containerPort }}
+            - containerPort: {{ . }}
           {{- end }}
           volumeMounts:
             - name: config
@@ -104,10 +105,10 @@ spec:
                 fieldRef:
                   apiVersion: v1
                   fieldPath: status.podIP
-{{- if .Values.command }}
+          {{- with .Values.command }}
           command:
-            {{- toYaml .Values.command | nindent 12 }}
-{{- end }}
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
           args:
             - --storage.path=/alertmanager
             - --config.file=/etc/alertmanager/alertmanager.yml
@@ -166,7 +167,7 @@ spec:
             optional: {{ . }}
             {{- end }}
         {{- end }}
-      {{- if .Values.persistence.enabled }}
+  {{- if .Values.persistence.enabled }}
   volumeClaimTemplates:
     - metadata:
         name: storage
@@ -183,7 +184,7 @@ spec:
         storageClassName: {{ .Values.persistence.storageClass }}
       {{- end }}
       {{- end }}
-{{- else }}
+  {{- else }}
         - name: storage
           emptyDir: {}
-{{- end -}}
+  {{- end }}
